src/main/java/org/banker/loan/resources/LoanResource.java

```diff
@@ -107,6 +110,36 @@ public Response  status(@PathParam("loanId") Long loanId, LoanStatus status, Uri
                 .build();
     }
 
+    @GET
+    @Path("/search")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(summary = "search", description = "Retrieve a data by name,id,email and number")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Retrieve of list of loan"),
+            @APIResponse(responseCode = "204", description = "No Content"),
+            @APIResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseDto search(
+            @QueryParam("searchValue") String searchValue, @QueryParam("page") int page, @QueryParam("size") int size) {
```

Comment:
Ensure input values are validated before processed. Are you handling the error response?

Reply:
added validation and error response for fields